Spec/RealtimeClientConnection.swift

```diff
@@ -2649,6 +2649,118 @@ class RealtimeClientConnection: QuickSpec {
+                                    let disconnectedAt = Date()
+                                    expect(client.connectionStateTtl).to(equal(customTtlInterval))
+                                    expect(client.maxIdleInterval).to(equal(customIdleInterval))
+                                    client.connection.once(.connecting) { _ in
+                                        let reconnectionInterval = Date().timeIntervalSince(disconnectedAt)
+                                        expect(reconnectionInterval).to(beGreaterThan(client.connectionStateTtl + client.maxIdleInterval))
+                                        client.connection.once(.connected) { _ in
+                                            expect(client.connection.id).toNot(equal(connectionId))
+                                            done()
+                                        }
+                                    }
+                                }
+                                client.onDisconnected()
+                            }
+                        }
+                    }
+                    // RTN15g3
+                    it("reattaches to the same channels after a new connection has been established") {
+                        client = AblyTests.newRealtime(options)
+                        client.connect()
+                        defer { client.close() }
+                        let channelName = "test-reattach-after-ttl"
+                        let channel = client.channels.get(channelName)
+                        
+                        waitUntil(timeout: testTimeout) { done in
+                            client.connection.once(.connected) { _ in
+                                connectionId = client.connection.id!
+                                client.connectionStateTtl = customTtlInterval
+                                client.maxIdleInterval = customIdleInterval
+                                channel.attach { error in
+                                    if let error = error {
+                                        fail(error.message)
+                                    }
+                                    expect(channel.state).to(equal(ARTRealtimeChannelState.attached))
+                                }
+                                client.connection.once(.disconnected) { _ in
+                                    client.connection.once(.connecting) { _ in
+                                        client.connection.once(.connected) { _ in
+                                            expect(client.connection.id).toNot(equal(connectionId))
+                                            channel.once(.attached) { _ in
+                                                done()
+                                            }
+                                        }
+                                    }
+                                }
+                                client.onDisconnected()
```

Comment:
Why is this done before the attach is completed?  The sequence of events for this test is wrong i.e. the initial `attach` may only happen once the connection becomes connected again, in fact that is very likely.

Reply:
@mattheworiordan I don't understand. Would you prefer to see this within the `attach` block and right after this assertion? https://github.com/ably/ably-ios/blob/develop/Spec/RealtimeClientConnection.swift#L2708

Reply:
Yes, exactly, otherwise how do we know that the channel attached first and thus gets reattached?